### DISCUSSION TITLE
about load rate

### DISCUSSION THREAD

**Discussion by lifengyu2005** [greg7mdp/parallel-hashmap#173] [Q&A]
We used 16 submaps to insert 2 * 1024 * 1024, and found that the loading rate was about 50%; We use a fixed number of items and do not insert items dynamically in the middle, but expect a higher load rate, otherwise memory will be wasted seriously; Are there any parameters that can be adjusted ？

**Comment by greg7mdp** [comment]
Hum, possibly you got unlucky. For a regular `flat_hash_map` the `load_factor` would be between 50% and 87%, but it grows in powers of two. So if you insert 2 * 1024 * 1024 elements, which is a power of two, you would typically hit a resize when it reached 2048 * 0.87 or about 1780 values, so now you have 2048 values in a table that would resize when you have inserted about 3563 values (4096 * 0.87).

For a `parallel_flat_hash_map`, the values should be similar, although they might differ a little bit.
